### PR TITLE
thread-safe wcs

### PR DESCRIPTION
**Pull request by jeff-regier**
Hi @kbarbary  --  We're getting segmentation faults from libwcs every time in our multi-threaded code without this lock. There may be a better way to fix the problem, but it's not obvious to me what that would be. (I haven't found a flag to compile reentrant version of libwcs.). I thought I'd submit this PR as a placeholder at least.

### CODE REVIEW

**Comment by jeff-regier** [comment]
Ah, Tony pointed this out an email thread:

From http://www.atnf.csiro.au/people/mcalabre/WCS/wcslib/threads.html
"The C code generated by Flex is not re-entrant. Flex does have the capacity for producing re-entrant scanners but they have a different API. This may be handled by a compile-time option in future but in the meantime calls to the header parsers should be serialized via a mutex."

**Comment by kbarbary** [comment]
For completeness, replicating discussion from the email thread:

> but in the meantime calls to the **header parsers** should be serialized via a mutex.

`wcss2p` isn't a header-parsing function so it should already be thread-safe according to the linked webpage above. We should put locks around the ccalls in the `WCS.from_header()` function though.

**Comment by jeff-regier** [comment]
If I serialize the calls to `wcspih()` and `wcsbth()`, I get a different error message from julia-debug than when I wasn't serializing anything.

```
signal (6): Aborted
while loading /global/homes/j/jregier/.julia/v0.5/Celeste/bin/celeste.jl, in expression starting on line 82
wcsulexe at /global/u2/j/jregier/WCS.jl/deps/src/wcslib-5.13/C/wcsulex.c:8406
wcsunitse at /global/u2/j/jregier/WCS.jl/deps/src/wcslib-5.13/C/wcsunits.c:117
wcs_types at /global/u2/j/jregier/WCS.jl/deps/src/wcslib-5.13/C/wcs.c:2299 [inlined]
wcsset at /global/u2/j/jregier/WCS.jl/deps/src/wcslib-5.13/C/wcs.c:1918
wcs_units at /global/u2/j/jregier/WCS.jl/deps/src/wcslib-5.13/C/wcs.c:2580 [inlined]
wcsset at /global/u2/j/jregier/WCS.jl/deps/src/wcslib-5.13/C/wcs.c:1923
wcss2p at /global/u2/j/jregier/WCS.jl/deps/src/wcslib-5.13/C/wcs.c:2871
#world_to_pix!#4 at /global/homes/j/jregier/.julia/v0.5/WCS/src/WCS.jl:600
wcss2p at /global/u2/j/jregier/WCS.jl/deps/src/wcslib-5.13/C/wcs.c:2871
#world_to_pix!#4 at /global/homes/j/jregier/.julia/v0.5/WCS/src/WCS.jl:600
```

Any ideas? Is `wcss2p` calling a header-parsing function around line 8406 of wcsulex.c?

File: src/WCS.jl [hunk 1]
```diff
@@ -12,6 +12,18 @@ using Compat
 import Compat.ASCIIString
 
 
+if VERSION > v"0.5.0-dev"
```
Comment (tkelman) [inline src/WCS.jl]: maybe better to use `isdefined(Base, :Threads)` for this so it's more easily bisect-able

**Pull request closed without merging**